test/shared/src/main/scala/zio/test/mock/Mock.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.mock
+
+import zio.internal.Executor
+import zio.stream.{ ZSink, ZStream }
+import zio.test.TestPlatform
+import zio.{ Has, Runtime, Tagged, URIO, URLayer, ZIO }
+
+/**
+ * A `Mock[R]` represents a mockable environenment `R`.
+ */
+abstract class Mock[R <: Has[_]: Tagged] { self =>
```

Comment:
Covariant?

Reply:
It would probably break things, but have not tested. I don't think it would bring any value though, at least in current implementation.